qf_lib/plotting/charts/waterfall_chart.py

```diff
@@ -0,0 +1,60 @@
+#     Copyright 2016-present CERN – European Organization for Nuclear Research
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+from typing import Tuple, Optional, List
+
+import numpy as np
+
+from qf_lib.containers.series.qf_series import QFSeries
+from qf_lib.plotting.charts.chart import Chart
+
+
+class WaterfallChart(Chart):
+    def __init__(self, data: QFSeries, title: Optional[str] = None, total: Optional[List] = None):
+        super().__init__()
```

Comment:
The usage of the Total is a little bit confusing in my opinion. We could put in the demo
`WaterfallChart(my_series, title="Waterfall Chart", total=['Total', 'Value 3'])` and weird things would appear.

I would suggest to make Total a boolean flag, and if its true - the new element is appended to the data series as 'Total'

Reply:
Now there's a function that enables us add the data as a 'Total' column (add_total), or flag one of the existing columns as 'Total' (flag_total)

![image](https://github.com/quarkfin/qf-lib/assets/41996369/d7037ccb-1d86-432b-b78f-bcad0d1f5fab)
![image](https://github.com/quarkfin/qf-lib/assets/41996369/9cea5aac-cdd4-43d7-971c-5f28d2f1732a)